include/cantera/thermo/CoverageDependentSurfPhase.h

```diff
@@ -0,0 +1,476 @@
+
+    //! Get the chemical potential vector. Units: J/kmol.
+    /*!
+     * \f[
+     *      \mu_k(T,\theta) = \mu^o_k(T,\theta) + RTln(\theta_k)
+     * \f]
+     */
+    virtual void getChemPotentials(double* mu) const;
+    //! @}
+
+    //! @name Methods calculating Phase thermodynamic properties
+    //! Phase properties are evaluated at \f$ T \text{ and } \theta \f$,
+    //! and thus are dependent both on temperature and coverage.
+
+    //! @{
+
+    //! Return the solution's molar enthalpy. Units: J/kmol
+    /*!
+     * \f[
+     *      \hat h(T,\theta) = \sum_k \theta_k \overline{h}_k(T,\theta)
+     * \f]
+     */
+    virtual double enthalpy_mole() const;
+
+    //! Return the solution's molar entropy. Units: J/kmol/K
+    /*!
+     * \f[
+     *      \hat s(T,\theta) = \sum_k \theta_k \overline{s}_k(T,\theta)
+     * \f]
+     */
+    virtual double entropy_mole() const;
+
+    //! Return the solution's molar cp. Units: J/kmol/K
+    /*!
+     * \f[
+     *      \hat{cp}(T,\theta) = \sum_k \theta_k \overline{cp}_k(T,\theta)
+     * \f]
+     */
+    virtual double cp_mole() const;
+    //! @}
+
+protected:
+    //! Temporary storage for the coverages.
+    mutable vector_fp m_cov;
+
+    //! Temporary storage for the coverage-dependent enthalpies.
+    mutable vector_fp m_h_cov;
+
+    //! Temporary storage for the coverage-dependent entropies.
+    mutable vector_fp m_s_cov;
+
+    //! Temporary storage for the coverage-dependent heat capacities.
+    mutable vector_fp m_cp_cov;
+
+    //! Temporary storage for the coverage-dependent chemical potentials.
+    mutable vector_fp m_mu_cov;
+
+    //! Temporary storage for the sum of reference state enthalpies and
+    //! coverage-dependent enthalpies.
+    mutable vector_fp m_enthalpy;
+
+    //! Temporary storage for the sum of reference state entropies and
+    //! coverage-dependent entropies.
+    mutable vector_fp m_entropy;
+
+    //! Temporary storage for the sum of reference state heat capacities and
+    //! coverage-dependent heat capacities.
+    mutable vector_fp m_heatcapacity;
+
+    //! Temporary storage for the sum of reference state chemical potentials
+    //! and coverage-dependent chemical potentials.
+    mutable vector_fp m_chempot;
+
+    //! Array of enthalpy and entropy coverage dependency parameters used in
+    //! the linear and polynomial dependency equations.
+    std::vector<PolynomialDependency> m_PolynomialDependency;
+
+    //! Array of enthalpy and entropy coverage dependency parameters used in
+    //! the piecewise-linear and interpolative dependency equations.
+    std::vector<InterpolativeDependency> m_InterpolativeDependency;
+
+    //! Array of heat capacity coverage dependency parameters.
+    std::vector<HeatCapacityDependency> m_HeatCapacityDependency;
+
+    //! Explicitly-specified k-j interaction parameters, to enable serialization.
+    //! For linear model. <name_k>: <name_j: PolynomialDependency_index>>.
```

Comment:
This formatting in the docstring causes Doxygen to generate warnings. I would write suggest writing this as something like
```suggestion
    //! For linear model. `m_indexmap_lin[name_k][name_j] = ` index into #m_PolynomialDependency.
```
Of course, this only matters if this variable is sticking around (per other comments).